lib/repository/data_repository.dart

```diff
@@ -13,25 +11,17 @@ class DataRepository {
 
   Future<RepoDataModel> getData(
       {required String repoName, required String sort, int page = 1}) async {
-    try {
-      // //エラーテスト用URL
-      // final apiUri = Uri.parse('https://httpstat.us/403');
-
-      final resUrl =
-          'https://api.github.com/search/repositories?q=$repoName&sort=$sort&page=$page&per_page=20';
-
-      final apiUrl = Uri.parse(Uri.encodeFull(resUrl));
-
-      http.Response response = await client.get(apiUrl);
-
-      if (response.statusCode == 200) {
-        final jsonData = json.decode(response.body);
-        return RepoDataModel.fromJson(jsonData);
-      } else {
-        throw UnknownException();
-      }
-    } on SocketException catch (_) {
-      throw NoInternetException();
-    }
+    // //エラーテスト用URL
+    // final apiUri = Uri.parse('https://httpstat.us/429');
+
+    final resUrl =
+        'https://api.github.com/search/repositories?q=$repoName&sort=$sort&page=$page&per_page=20';
+
+    final apiUrl = Uri.parse(Uri.encodeFull(resUrl));
+    http.Response response =
+        await client.get(apiUrl).timeout(const Duration(seconds: 30));
```

Comment:
応答が遅い時や、インターネット完全に切れていない状況だったり、弱い状況だったりでエラーに拾われない時のために、負荷を減らすため、ユーザーの時間を無駄にしないためにタイムアウトを設定しています。

> Most requests to external servers should have a timeout attached, in case the server is not responding in a timely manner. By default, requests do not time out unless a timeout value is set explicitly. Without a timeout, your code may hang for minutes or more.
外部サーバーへのリクエストのほとんどは、サーバーがタイムリーに応答しない場合に備えて、タイムアウトを付けるべきです。デフォルトでは、タイムアウト値が明示的に設定されない限り、リクエストはタイムアウトしません。タイムアウトを設定しないと、コードが何分もハングアップする可能性があります。



from http://docs.python-requests.org/en/latest/user/advanced/#timeouts

pythonのdocですが、このような理由です。
標準のhttpのタイムアウトを使うと、わざわざ書く必要ないよと言うことであれば教えていただきたいです。